Fix safer C++ static analyzer warnings in WebFullScreenManager.cpp

rniwa opened the pull request:
#### e6d5de7d65d7a14b1d0345a17c61e837fd8d751f
<pre>
Fix safer C++ static analyzer warnings in WebFullScreenManager.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=301372">https://bugs.webkit.org/show_bug.cgi?id=301372</a>

Reviewed by Jer Noble.

Addressed the static analyzer warnings by deploying more smart pointers.

* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp:
(WebKit::WebFullScreenManager::willExitFullScreen):

Canonical link: <a href="https://commits.webkit.org/302062@main">https://commits.webkit.org/302062@main</a>
</pre>

jernoble reviewed (approved):
r=me, but I think the change to the expectations file may need another revision.

rniwa committed:
Fix safer C++ static analyzer warnings in WebFullScreenManager.cpp
https://bugs.webkit.org/show_bug.cgi?id=301372

Reviewed by Jer Noble.

Addressed the static analyzer warnings by deploying more smart pointers.

* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp:
(WebKit::WebFullScreenManager::willExitFullScreen):

Canonical link: https://commits.webkit.org/302062@main
